src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
 
 } // namespace
 
+void ShaderNetworkAlgo::convertToOSLConventions( ShaderNetwork *network, int oslVersion )
+{
+	std::map< IECore::InternedString, std::pair< IECore::InternedString, size_t > > currentSplineArrayAdapters;
+
+	for( const auto &s : network->shaders() )
+	{
+		if( !boost::starts_with( s.second->getType(), "osl:" ) )
+		{
+			continue;
+		}
+
+		const auto &origParameters = s.second->parameters();
+		bool hasSplines = false;
+		for( const auto &i : origParameters )
+		{
+			if(
+				runTimeCast<const SplinefColor3fData>( i.second.get() ) ||
+				runTimeCast<const SplineffData>( i.second.get() )
+			)
+			{
+				hasSplines = true;
+				break;
+			}
+		}
+
+		if( !hasSplines )
+		{
+			continue;
+		}
```

Comment:
What's the purpose of this pre-check? Is it just to avoid the `new CompoundData` when we don't need it? Thinking forwards to your todo on L916, this `continue` seems a bit incompatible with integrating the other conversions. I wonder if the loop over the shaders could call `convertSplines()`, `convertComponents()` etc, passing them each a lambda which will acquire the new parameters on demand? Not sure how important this is now though...